src/main/java/net/dv8tion/jda/core/JDABuilder.java

```diff
@@ -538,7 +557,8 @@ public JDA buildAsync() throws LoginException, IllegalArgumentException, RateLim
                 .setCacheGame(game)
                 .setCacheIdle(idle)
                 .setCacheStatus(status);
-        jda.login(token, shardInfo, reconnectQueue);
+        IGatewayProviderFactory gatewayProviderFactory = this.gatewayProviderFactory == null ? new DefaultGatewayProviderFactory() : this.gatewayProviderFactory;
```

Comment:
This check should happen inside the WebSocketClient constructor

Reply:
I would actually like to see this handled in the same way that the `eventManager` and the `audioSendFactory` are handled above.